mongo/integration/unified_spec_test.go

```diff
@@ -194,7 +195,13 @@ func runSpecTestFile(t *testing.T, specDir, fileName string) {
 	assert.Nil(t, err, "unable to unmarshal spec test file at %v: %v", filePath, err)
 
 	// create mtest wrapper and skip if needed
-	mt := mtest.New(t, mtest.NewOptions().RunOn(testFile.RunOn...).CreateClient(false))
+	mtOpts := mtest.NewOptions().
+		RunOn(testFile.RunOn...).
+		CreateClient(false)
+	if specDir == "atlas-data-lake-testing" {
+		mtOpts.AtlasDataLake(true)
```

Comment:
Matching on the directory name kinda sucks, but these tests fail if run against a non-ADL server because the seed data isn't present. We might be able to do this by putting the top-level test function in a separate file like `data_lake_spec_test.go`, not modifying the `directories` slice in this file to include the `atlas-data-lake-testing` directory, and having the other file call this function directly. We do something similar for [client-side encryption spec tests](https://github.com/mongodb/mongo-go-driver/blob/1962fcb241bc7bc879e49ce18034a74e55f32c3d/mongo/integration/client_side_encryption_spec_test.go#L20) because they are only run if the `cse` build tag is enabled.

Does anyone have a preference?